src/runtime/methodbinder.cs

```diff
@@ -378,6 +378,7 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                 var outs = 0;
                 var margs = TryConvertArguments(pi, paramsArray, args, pynargs, kwargDict, defaultArgList,
                     needsResolution: _methods.Length > 1,  // If there's more than one possible match.
+                    setError: _methods.Length == 1 && !isGeneric,   // True when there are no alternatives.
```

Comment:
Since you set an error here, it must be cleared unless propagated (e.g. when next overload is tried).

Reply:
The condition here is meant to exclude all cases where the error might need to be cleared, such as another overload.

Reply:
Since you can set the `__cause__`, I see no reason to ever pass `setError: false`. If there were > 1 overload, and one matched, you can just clear the error.